PetGame/Services/PetService.cs

```diff
@@ -200,5 +221,244 @@ public bool DeletePet(ulong petId, ulong ownerId)
+
+                //check the number of minute since the last activity
+                int MinutesSinceLastActivity = 0;
+                MinutesSinceLastActivity = DateTime.Now.Minute - LastActivity.Timestamp.Minute;
+
+                //if it's been more than 5 minutes, the user can perform another action
+                //otherwise, leave the default 5 min value in place
+                if (MinutesSinceLastActivity > 5)
+                {
+                    MinutesTillNextAction = 0;
+                }
+            }
+            return MinutesTillNextAction;
+        }
+
+        private double CalculateHunger(IEnumerable<Activity> PastActivities, double hungerPercentage, int HoursToCheck)
+        {
+            List<Activity> Activities = (List<Activity>)PastActivities;
+
+            if (Activities.Count == 0)
+            {
+                hungerPercentage -= (HungerDecreasePerHour * HoursToCheck);
+            }
+            else
+            {
+                //check the type of activity and subtract percentage accordingly
+                foreach (Activity Activity in PastActivities)
+                {
+                    //if a Feeding activity has occured, increase hunger
+                    if (Activity.Type.Equals(ActivityType.Feeding))
+                    {
+                        hungerPercentage = ModifyPercentage(hungerPercentage, FeedingIncrease, 'a');
+                    }
+                    //if a Training event has occured, subtract
+                    else if (Activity.Type.Equals(ActivityType.Training))
+                    {
+                        hungerPercentage = ModifyPercentage(hungerPercentage, TrainingDecrease, 's');
+                    }
+                    //if a Race event has occurred
+                    else if (Activity.Type.Equals(ActivityType.Race))
+                    {
+                        hungerPercentage = ModifyPercentage(hungerPercentage, RaceDecrease, 's');
+                    }
+                    //else, decrease hunger
+                    else
+                    {
+                        hungerPercentage = ModifyPercentage(hungerPercentage, HungerDecreasePerHour, 's');
+                    }
+                }
+            }
+            return hungerPercentage;
+        }
+
+        private double ModifyPercentage(double percentage, double changeFactor, char operation)
+        {
+            //a stands for add
+            if (operation == 'a')
+            {
+                if (percentage < MaximumPercentage &&
+                            ((percentage += changeFactor) <= MaximumPercentage))
+                {
+                    percentage += changeFactor;
+                }
+                else
+                {
+                    percentage = MaximumPercentage;
+                }
+                return percentage;
+            }
+            //s stands for subtract
+            else if (operation == 's')
```

Comment:
Also, why can't you just perform the action normally, then call `ConstrainPercentages` on the result?